src/Plugins/Artemis.Plugins.Modules.General/DataModels/Windows/WindowDataModel.cs

```diff
@@ -14,6 +20,28 @@ public WindowDataModel(Process process)
 
             // Accessing MainModule requires admin privileges, this way does not
             ProgramLocation = process.GetProcessFilename();
+
+            // Get Icon colors
+            if(File.Exists(ProgramLocation)) 
+            {
+                MemoryStream mem = new MemoryStream();
+                Icon.ExtractAssociatedIcon(ProgramLocation).Save(mem);
+                mem.Seek(0, SeekOrigin.Begin);
+                var skbm = SKBitmap.Decode(mem);
+                mem.Close();
+
+                IColorQuantizerService colorQuantizer = new ColorQuantizerService();
```

Comment:
This is a mistake I overlooked in the Quantizer PR, you shouldn't be able to create your own instances of services. I fixed that in https://github.com/Artemis-RGB/Artemis/commit/76d426c3d2734aaa5dec87be92bc3ad688fb1c94. Instead, you should inject that service into the plugin and let Artemis manage its lifecycle.

```cs
    public class GeneralModule : ProfileModule<GeneralDataModel>
    {
        private readonly IColorQuantizerService _quantizerService;

        public GeneralModule(IColorQuantizerService quantizerService)
        {
            _quantizerService = quantizerService;
        }

        ...

        public void UpdateCurrentWindow()
        {
            int processId = WindowUtilities.GetActiveProcessId();
            if (DataModel.ActiveWindow == null || DataModel.ActiveWindow.Process.Id != processId)
                DataModel.ActiveWindow = new WindowDataModel(Process.GetProcessById(processId), _quantizerService);
            if (DataModel.ActiveWindow != null && string.IsNullOrWhiteSpace(DataModel.ActiveWindow.WindowTitle))
                DataModel.ActiveWindow.WindowTitle = Process.GetProcessById(WindowUtilities.GetActiveProcessId()).MainWindowTitle;
        }
    }
```